Documentation/network/egress-gateway.rst

```diff
@@ -481,3 +481,12 @@ will be the ``egressIP`` from the policy, if specified).
 If the egress list shown does not contain entries as expected to match your
 policy, check that the pod(s) and egress node are labeled correctly to match
 the policy selectors.
+
+Troubleshooting Guide
```

Comment:
Separating troubleshooting guide from the main EG page makes sense to me, but a TOC here can be a bit confusing I think, as there is nothing to list really, it's just one link.
![Screenshot from 2024-07-09 11-37-42](https://github.com/cilium/cilium/assets/3199463/ec8f974b-5487-44f0-880a-be16ea254712)

There is also a general troubleshooting page in `Documentation/operations/troubleshooting.rst`. How about we:
1) add "Egress Gateway Troubleshooting" section to the main troubleshooting page
2) just link to this section here without a TOC?

Reply:
Good point.  I want to keep this topic in a seperate page from the general EGW page, as it feels pretty esoteric for the general EGW guide.

I did a direct link instead of the TOC. As well, I added the separate troubleshooting page as EGW "Advanced Troubleshooting" since the linter complains if there are pages not included in the root level TOC:

![image](https://github.com/cilium/cilium/assets/5881740/a038ef6d-6f00-4691-99e9-202617010774)

Reply:
I do wonder if having EGW under "External networking" even makes sense.  EGW is a feature in itself - it may make sense to split the EGW articles into their own dedicated section:

i.e. the following structured

* External networking
    * Setting up Support for External Workloads (beta)
* Egress Gateway
    * Getting started
    * Advanced Troubleshooting

@julianwiedmann wdyt?

Reply:
Actually, I'm sort of leaning towards moving to a dedicated EGW section, see:

Index Page:

![image](https://github.com/cilium/cilium/assets/5881740/e3f8a6e5-4e5b-47b2-bfd7-661db6240f7d)

And the new EGW sub-index:

![image](https://github.com/cilium/cilium/assets/5881740/e6a5500e-f0df-48e0-9189-c313c1859959)

Reply:
this makes a lot of sense. these two should never have been grouped together